packages/components/inputs/search-select-input/src/search-select-input.tsx

```diff
@@ -10,235 +10,282 @@ import {
-   * <br />
-   * Signature: `(inputValue, action) => void`
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  onInputChange: PropTypes.func,
+  onInputChange?: ReactSelectAsyncProps['onInputChange'];
   /**
    * Select the currently focused option when the user presses tab
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  tabSelectsValue: PropTypes.bool,
+  tabSelectsValue?: ReactSelectAsyncProps['tabSelectsValue'];
   /**
    * Function that returns a promise, which is the set of options to be used once the promise resolves.
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  loadOptions: PropTypes.func.isRequired,
+  loadOptions: ReactSelectAsyncProps['loadOptions'];
   /**
    * The text shown while the options are being loaded
    */
-  loadingMessage: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
+  loadingMessage?: string | (() => string);
   /**
    * If cacheOptions is truthy, then the loaded data will be cached. The cache will remain until cacheOptions changes value.
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  cacheOptions: PropTypes.any,
+  cacheOptions?: ReactSelectAsyncProps['cacheOptions'];
   /**
    * Custom method to filter whether an option should be displayed in the menu
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  filterOption: PropTypes.func,
+  filterOption?: ReactSelectAsyncProps['filterOption'];
   /**
    * The style of the an option in the dropdown menu. It could be single lined option or an option with more and custom info
    */
-  optionType: PropTypes.oneOf([
-    'single-property',
-    'double-property',
-    'multiple-properties',
-  ]),
+  optionType?: 'single-property' | 'double-property' | 'multiple-properties';
+};
+
+type TOptionInnerPropsData = {
+  label?: string;
+  key?: string;
+  id?: string;
+};
+
+type TOptionInnerProps = {
+  data?: TOptionInnerPropsData;
+} & OptionProps;
+
+const SearchSelectInput = (props: TSearchSelectInputProps) => {
+  const intl = useIntl();
+  const noOptionsMessage =
+    props.noOptionsMessage ||
+    (() => intl.formatMessage(messages.noOptionsMessage));
+  const loadingMessage =
+    props.loadingMessage || intl.formatMessage(messages.loadingOptionsMessage);
+  const placeholder =
+    props.placeholder || intl.formatMessage(messages.placeholderMessage);
+  const optionType = props.optionType;
+
+  const components = useMemo(() => {
+    return {
+      Option: (optionInnerProps: TOptionInnerProps) => (
+        <CustomSelectInputOption
+          {...optionInnerProps}
+          optionType={optionType}
+          optionInnerProps={optionInnerProps}
+        />
+      ),
+      ...props.components,
+      DropdownIndicator: SearchIconDropdownIndicator,
+    };
+  }, [props.components, optionType]);
```

Comment:
nit: Maybe using the condensed format?

```suggestion
const components = useMemo(() => ({  
  Option: (optionInnerProps: TOptionInnerProps) => (
    <CustomSelectInputOption
      {...optionInnerProps}
      optionType={optionType}
      optionInnerProps={optionInnerProps}
    />
  ),
  ...props.components,
  DropdownIndicator: SearchIconDropdownIndicator,
}), [props.components, optionType]);
```